AppCenter/AppCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -264,100 +275,172 @@ + (void)enableAutoVacuumInOpenedDatabase:(void *)db {
    */
   if (autoVacuumDisabled) {
     MSLogDebug([MSAppCenter logTag], @"Vacuuming database and enabling auto_vacuum");
-    [MSDBStorage executeNonSelectionQuery:@"PRAGMA auto_vacuum = FULL; VACUUM" inOpenedDatabase:db];
+
+    // We use a slection query here because pragma set returns a value.
+    [MSDBStorage executeSelectionQuery:@"PRAGMA auto_vacuum = FULL;" inOpenedDatabase:db withValues:nil];
+    [MSDBStorage executeSelectionQuery:@"VACUUM;" inOpenedDatabase:db withValues:nil];
   }
 }
 
-- (NSUInteger)countEntriesForTable:(NSString *)tableName condition:(nullable NSString *)condition {
+- (NSUInteger)countEntriesForTable:(NSString *)tableName condition:(nullable NSString *)condition withValues:(nullable NSArray *)values {
   NSMutableString *countLogQuery = [NSMutableString stringWithFormat:@"SELECT COUNT(*) FROM \"%@\" ", tableName];
   if (condition.length > 0) {
     [countLogQuery appendFormat:@"WHERE %@", condition];
   }
-  NSArray<NSArray<NSNumber *> *> *result = [self executeSelectionQuery:countLogQuery];
+  NSArray<NSArray<NSNumber *> *> *result = [self executeSelectionQuery:countLogQuery withValues:values];
   return (result.count > 0) ? result[0][0].unsignedIntegerValue : 0;
 }
 
++ (int)executeNonSelectionQuery:(NSString *)query inOpenedDatabase:(void *)db {
+  return [self executeNonSelectionQuery:query inOpenedDatabase:db withValues:nil];
+}
+
 - (int)executeNonSelectionQuery:(NSString *)query {
+  return [self executeNonSelectionQuery:query withValues:nil];
+}
+
+- (int)executeNonSelectionQuery:(NSString *)query withValues:(nullable NSArray *)values {
   return [self executeQueryUsingBlock:^int(void *db) {
-    return [MSDBStorage executeNonSelectionQuery:query inOpenedDatabase:db];
+    return [MSDBStorage executeNonSelectionQuery:query inOpenedDatabase:db withValues:values];
   }];
 }
 
-+ (int)executeNonSelectionQuery:(NSString *)query inOpenedDatabase:(void *)db {
-  char *errMsg = NULL;
-  int result = sqlite3_exec(db, [query UTF8String], NULL, NULL, &errMsg);
-  if (result == SQLITE_CORRUPT || result == SQLITE_NOTADB) {
-    MSLogError([MSAppCenter logTag], @"A database file is corrupted: %d - %@", result, [NSString stringWithUTF8String:errMsg]);
-  } else if (result == SQLITE_FULL) {
-    MSLogDebug([MSAppCenter logTag], @"Query failed with error: %d - %@", result, [NSString stringWithUTF8String:errMsg]);
-  } else if (result != SQLITE_OK) {
-    MSLogError([MSAppCenter logTag], @"Query \"%@\" failed with error: %d - %@", query, result, [NSString stringWithUTF8String:errMsg]);
++ (int)executeNonSelectionQuery:(NSString *)query inOpenedDatabase:(void *)db withValues:(nullable NSArray *)values {
+  return
+      [MSDBStorage executeQuery:query
+               inOpenedDatabase:db
+                     withValues:values
+                     usingBlock:^(void *statement) {
+                       int stepResult = sqlite3_step(statement);
+                         if (stepResult == SQLITE_DONE) {
```

Comment:
Looks like this is indented too far, should line up with the line above it

Reply:
@annakocheshkova please just run code formatter